.github/workflows/ci.yml

```diff
@@ -115,28 +115,26 @@ jobs:
         cmake --build $GITHUB_WORKSPACE/test/install/build --config ${{ matrix.type.name == 'Debug' && 'Debug' || 'Release' }}
 
   format:
-    name: Formatting on ${{ matrix.platform.name }}
+    name: Formatting with ${{ matrix.platform.name }}
     runs-on: ${{ matrix.platform.os }}
 
     strategy:
       fail-fast: false
       matrix:
         platform:
-        - { name: Windows, os: windows-2022,  executable: clang-format }
-        - { name: Linux,   os: ubuntu-latest, executable: clang-format-12 }
-        - { name: macOS,   os: macos-12,      executable: clang-format }
+        - { name: clang-format-12, os: ubuntu-22.04 }
+        - { name: clang-format-13, os: ubuntu-22.04 }
+        - { name: clang-format-14, os: ubuntu-22.04 }
```

Comment:
This can be simpler. For example:
```yaml
matrix:
   version: [12, 13, 14]
```
The os is constant `ubuntu-22.04` and the package name to install or executable name is `clang-format-${{matrix.version}}`